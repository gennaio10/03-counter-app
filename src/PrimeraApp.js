import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, perfil, empresa = "Laura inc" }) => {
  const persona = {
    nombres: "Iron man",
    edad: 34,
  };

  return (
    <>
      <h1>Bienvenido...</h1>
      <p>{saludo}</p>
      <p id="tperfil" >{perfil}</p>
      <p>{empresa}</p>
      <pre>{JSON.stringify(persona, null, 3)}</pre>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  perfil: "Perfil por defecto",
};

export default PrimeraApp;
