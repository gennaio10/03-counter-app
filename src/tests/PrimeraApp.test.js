
import React from "react";
import '@testing-library/jest-dom';
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("Pruebas de <PrimeraApp/>", () => { 
  test(' debe mostrar el mensaje "Hola, soy Goku"', () => {
    const saludo = "Hola, soy Goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test(' debe mostrar el perfil correcto', () => {
    const saludo = "Hola, soy Goku";
    const perfil = "Administrador";
    
    const wrapper = shallow(<PrimeraApp saludo={saludo} 
      perfil={perfil}
       />);

    const textoPerfil = wrapper.find('#tperfil').text();
    expect(textoPerfil).toBe(perfil);
  });


});
