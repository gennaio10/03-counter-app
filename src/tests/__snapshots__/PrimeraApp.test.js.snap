// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <PrimeraApp/>  debe mostrar el mensaje "Hola, soy Goku" 1`] = `
<Fragment>
  <h1>
    Bienvenido...
  </h1>
  <p>
    Hola, soy Goku
  </p>
  <p
    id="tperfil"
  >
    Perfil por defecto
  </p>
  <p>
    Laura inc
  </p>
  <pre>
    {
   "nombres": "Iron man",
   "edad": 34
}
  </pre>
</Fragment>
`;

exports[`Pruebas de <PrimeraApp/>  debe mostrar el mensaje "Hola, soy Goku" 2`] = `
<Fragment>
  <h1>
    Bienvenido...
  </h1>
  <p>
    Hola, soy Goku
  </p>
  <p>
    Administrador
  </p>
  <p>
    Laura inc
  </p>
  <pre>
    {
   "nombres": "Iron man",
   "edad": 34
}
  </pre>
</Fragment>
`;
